projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-uploader/index.native.js

```diff
@@ -0,0 +1,81 @@
+/**
+ * External dependencies
+ */
+import { BlockIcon, MediaPlaceholder } from '@wordpress/block-editor';
+import { useDispatch } from '@wordpress/data';
+import { useCallback, useState } from '@wordpress/element';
+import { __ } from '@wordpress/i18n';
+import { store as noticesStore } from '@wordpress/notices';
+import { isURL, getProtocol } from '@wordpress/url';
+/**
+ * Internal dependencies
+ */
+import { buildVideoPressURL } from '../../../../../lib/url';
+import { VIDEOPRESS_VIDEO_ALLOWED_MEDIA_TYPES } from '../../constants';
+import { description, title } from '../../index';
+import { VideoPressIcon } from '../icons';
+import UploadProgress from './uploader-progress';
+import './style.scss';
+
+const VideoPressUploader = ( { handleDoneUpload } ) => {
+	const [ uploadFile, setFile ] = useState( null );
+	const [ isUploadingInProgress, setIsUploadingInProgress ] = useState( false );
+	const { createErrorNotice } = useDispatch( noticesStore );
+
+	const onSelectURL = useCallback(
+		( videoSource, id ) => {
+			// If the video source is a VideoPress URL, we can use it directly.
+			const { guid, url: videoPressURL } = buildVideoPressURL( videoSource );
+			if ( ! guid ) {
+				createErrorNotice( __( 'Invalid VideoPress URL', 'jetpack-videopress-pkg' ) );
+				return;
+			}
+			handleDoneUpload( { guid, src: videoPressURL, id } );
+		},
+		[ createErrorNotice, handleDoneUpload ]
+	);
+
+	const onSelectVideo = useCallback(
+		media => {
```

Comment:
I noticed in the web version the upload is limited to the [first file if multiple are passed in](https://github.com/Automattic/jetpack/blob/trunk/projects/packages/videopress/src/client/block-editor/blocks/video/components/videopress-uploader/index.js#L165). Do we need something similar here to limit the uploads to 1 ?

Reply:
We only support uploading one file at a time, so initially, we don't need to check if multiple files are passed. I did a quick check and to enable multiple files, we'd need to pass the [prop `multiple` in the `MediaPlaceholder` component](https://github.com/WordPress/gutenberg/blob/a2b1c1147b9bb8e53a96908625476ab9e5405be2/packages/block-editor/src/components/media-placeholder/index.native.js#L50:L50).

Reply:
Cool, just making sure. I too noticed I was only able to select one video at a time.